### PR TITLE
test: recording in every format

### PR DESCRIPTION
**Pull request by blackxored**
Test that we record with no errors. Also tests every format. 
Working on verifying that the files are actually valid video files, it seems everything under the sun that could do this requires binaries/bindings. I'm still exploring here, would probably go with `mmmagic`.

### CODE REVIEW

File: test/recording.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    await focusChild(t);
+    await app.client.click('li=Output');
+    await setFormDropdown(t, 'Recording Format', format);
+    await app.client.click('button=Done');
+    await focusMain(t);
+
+    // Start recording
+    await app.client.click('.record-button');
+    await sleep(2000);
```
Comment (holiber) [inline test/recording.ts]: Use `waitForVisible` if it's possible instead `sleep`. You added `(2000 + 1000) * 6` ms to this tests with the `sleep` method

File: test/recording.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import * as path from 'path';
+import { tmpdir } from 'os';
+import { mkdtemp, readdir } from 'fs-extra';
+import { focusChild, focusMain, test, useSpectron } from './helpers/spectron';
+import { setFormDropdown, setFormInput } from './helpers/spectron/forms';
+import { sleep } from './helpers/sleep';
+
+useSpectron({ appArgs: '--nosync' });
```
Comment (holiber) [inline test/recording.ts]: '--nosync' is not necessary here, because this test doesn't use authorization

File: test/recording.ts (hunk 1 above)
Comment (blackxored) [reply test/recording.ts]: We're recording 2 sec videos, hence the sleep, does that make sense?

File: test/recording.ts (hunk 2 above)
Comment (blackxored) [reply test/recording.ts]: Got it. Thanks.

**Pull request merged**